iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -33,6 +32,12 @@
 using Microsoft.Azure.Devices.Shared;
 using Newtonsoft.Json;
 
+#if NET5_0
```

Comment:
This is getting quite messy. You might consider making a class somewhere that abstracts these changes away (calling TrySetResult or TryComplete).

Reply:
Yeah, I can't say that I like it the way it is now. I am planning on removing dependency on DotNetty's TCS (no params variant) and bring that into our library directly.